models/DataObject/ClassDefinition/CustomLayout.php

```diff
@@ -268,8 +268,17 @@ public function getDefinitionFile()
      */
     public static function cleanupForExport(&$data)
     {
-        if (isset($data->fieldDefinitionsCache)) {
-            unset($data->fieldDefinitionsCache);
+        if (method_exists($data, 'resolveBlockedVars')) {
```

Comment:
Maybe we should go for an interface here? Or we just add it the the abstract data-type?

Reply:
Yes, definitely use an interface here

Reply:
Added VarExportInterface for Data-types. please see https://github.com/pimcore/pimcore/pull/8200/commits/65a78b0aa1242b32cc94d14a2669b572c79dc585